analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/HashesShouldHaveUnpredictableSaltBase.cs

```diff
@@ -0,0 +1,29 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
+
+public abstract class HashesShouldHaveUnpredictableSaltBase : SymbolicRuleCheck
+{
+    protected const string DiagnosticId = "S2053";
+    protected const string MessageFormat = "{0}";
+    protected const string MakeSaltUnpredictableMessage = "Make this salt unpredictable.";
+    protected const string MakeThisSaltLongerMessage = "Make this salt at least 16 bytes.";
```

Comment:
For these two, `private` should be enough - in the next PR
```suggestion
    private const string MakeSaltUnpredictableMessage = "Make this salt unpredictable.";
    private const string MakeThisSaltLongerMessage = "Make this salt at least 16 bytes.";
```